### PR TITLE
Placement update/delete

### PR DESCRIPTION
**Pull request by itdove**
Signed-off-by: Dominique Vernier <dvernier@redhat.com>

https://github.com/stolostron/backlog/issues/22874
https://github.com/stolostron/backlog/issues/22875
https://github.com/stolostron/backlog/issues/22882

### CODE REVIEW

File: controllers/placementdecision/placementdecision_controller.go [hunk 1]
```diff
@@ -124,15 +125,16 @@ func (r *PlacementDecisionReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 }
 
-func (r *PlacementDecisionReconciler) processPlacementDecisionUpdate(placement *clusterv1alpha1.Placement) (ctrl.Result, error) {
```
Comment (robinbobbitt) [inline controllers/placementdecision/placementdecision_controller.go:127]: This file should really be renamed to placement_controller and all of the various incorrect comments should be cleaned up. It's a bit hard to wrap your head around this file and what it is and is not for in its current state. Not suggesting for this PR. :-)

Comment (robinbobbitt) [inline controllers/placementdecision/placementdecision_controller.go:127]: This file should really be renamed to placement_controller and all of the various incorrect comments should be cleaned up. It's a bit hard to wrap your head around this file and what it is and is not for in its current state. Not suggesting for this PR. :-)

File: controllers/placementdecision/placementdecision_controller.go [hunk 2]
```diff
@@ -276,22 +261,13 @@ func (r *PlacementDecisionReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
 	placementPredicate := predicate.Predicate(predicate.Funcs{
 		GenericFunc: func(e event.GenericEvent) bool { return false },
 		DeleteFunc: func(e event.DeleteEvent) bool {
 			// Filter only if obj is placement
 			switch e.Object.(type) {
 			case *clusterv1alpha1.Placement:
-				return annotationPlacementFilter(e.Object)
+				return isLinkedToStrategy(e.Object)
 			default:
 				return true
```
[the remark this answers is not in the document]
Comment (itdove) [reply controllers/placementdecision/placementdecision_controller.go:272]: The method a bit below https://github.com/itdove/idp-mgmt-operator/blob/c3414572b3ecc464a54b49f13e2421e0e232493a/controllers/placementdecision/placementdecision_controller.go#L298 WithEventFilter() gets events for all objects.
and so any object in the queue will use that filter and so as we are watching placement, Authrealm, Dexclient event, I have to filter only events for the placement.

[the remark this answers is not in the document]
Comment (itdove) [reply controllers/placementdecision/placementdecision_controller.go:272]: Maybe the WithEventFilter() should be placed at the end instead than just after the "For" for clarity reason.

[the remark this answers is not in the document]
Comment (robinbobbitt) [reply controllers/placementdecision/placementdecision_controller.go:272]: Ah, I see. My eyes jumped over WithEventFilter and I had assumed it was specific to Placement. Why not just pass it in as second argument of the For?

File: controllers/placementdecision/placementdecision_controller.go [hunk 3]
```diff
@@ -96,6 +96,7 @@ func (r *PlacementDecisionReconciler) Reconcile(ctx context.Context, req ctrl.Re
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
+			// Delete the strategy placement linked to that placement
```
[the remark this answers is not in the document]
Comment (itdove) [reply controllers/placementdecision/placementdecision_controller.go:99]: removed

File: controllers/placementdecision/placementdecision_controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (itdove) [reply controllers/placementdecision/placementdecision_controller.go]: Done

**Review by robinbobbitt — APPROVED** [review]
Thanks @itdove !

**Pull request merged**